src/corona_hakab_model/supervisor.py

```diff
@@ -258,27 +264,64 @@ def publish(self):
 
 
 class TabularSupervisable(Supervisable):
-    def __init__(self, interval: int):
+    def __init__(self, filter_fn: Callable[["manager.SimulationManager"], bool] = None):
         self.data = []
         self.sampling_days = []
-        self.interval = interval
+        self.filter = filter_fn
 
     def names(self):
-        return [f"t0 +{d} days" for d in self.sampling_days]
+        return [f"day {d}" for d in self.sampling_days]
 
     @abstractmethod
     def get(self, manager: "manager.SimulationManager"):
         pass
 
     def snapshot(self, manager: "manager.SimulationManager"):
-        if manager.current_step % self.interval == 0:
+        if self.filter is None or self.filter(manager):
             self.data.append(self.get(manager))
             self.sampling_days.append(manager.current_step)
 
     def publish(self):
         return {name_i: data_i for name_i, data_i in zip(self.names(), self.data)}
 
 
+class PeriodicReportSupervisable(TabularSupervisable):
+    def __init__(self, interval):
+        super().__init__(lambda manager: manager.current_step % interval == 0)
+
+    @abstractmethod
+    def get(self, manager: "manager.SimulationManager"):
+        pass
+
+
+class _AppliedPolicyReportSupervisable(TabularSupervisable):
+    def __init__(self):
+        super().__init__(_AppliedPolicyReportSupervisable.applied_policy_filter)
+
+    @staticmethod
+    def applied_policy_filter(manager: "manager.SimulationManager"):
+        return manager.policy_manager.daily_affected_circles is not None
+
+    def name(self) -> str:
+        return "policy_activation_stats"
+
+    def get(self, manager: "manager.SimulationManager"):
+        policy_message_to_circles_dict = manager.policy_manager.daily_affected_circles
+        affected_circles_report = {
+            'policy_type': [], 'circle_size': [], 'circle_kind': [], 'circle_connection_type': []
+        }
+        for message, circles in policy_message_to_circles_dict.items():
+            self.update_affected_circles_report(affected_circles_report, circles, message)
```

Comment:
I am surprised this compiles, its a static method you should call _AppliedPolicyReportSupervisable.method

Reply:
I fixed the function's call in a new commit